socorro/unittest/external/rabbitmq/test_crashstorage.py

```diff
@@ -149,19 +148,21 @@ def test__ack_crash(self):
 
         connection.channel.basic_ack.assert_called_once_with(delivery_tag=1)
 
-    def test_ack_crash_fails_gracefully(self):
+    def test_ack_crash_fails_gracefully(self, caplogpp):
+        caplogpp.set_level('WARNING')
+
         config = self._setup_config()
-        config.logger = Mock()
         crash_store = RabbitMQCrashStorage(config)
         crash_store.acknowledgment_queue.put('b2')
         crash_store._consume_acknowledgement_queue()
 
-        config.logger.warning.assert_called_once_with(
-            'RabbitMQCrashStorage tried to acknowledge crash %s'
-            ', which was not in the cache',
-            'b2',
-            exc_info=True
+        recs = [(rec.message, rec.exc_info) for rec in caplogpp.records]
```

Comment:
I kind of hate how caplog isn't helpful in regards to asserting things about the logged messages. I was tossing around wrapping that up with caplogpp into a library similar to the mock logger I wrote in Antenna:

https://github.com/mozilla-services/antenna/blob/master/testlib/loggingmock.py

I'll toss that in my list of things to think about.